version.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/urfave/cli"
+)
+
+// version will be populated by the Makefile, read from
+// VERSION file of the source code.
+var version = ""
+
+// gitCommit will be the hash that the binary was built from
+// and will be populated by the Makefile
+var gitCommit = ""
+
+var versionCommand = cli.Command{
+	Name:        "version",
+	Usage:       "output the runtime version",
+	Description: `The version command outputs the runtime version.`,
+	Action:      printVersion,
+}
+
+func printVersion(context *cli.Context) (err error) {
+	if version == "" {
```

Comment:
On Tue, Aug 23, 2016 at 11:09:30AM -0700, Michael Crosby wrote:
> > +// VERSION file of the source code.
> > +var version = ""
> > +
> > +// gitCommit will be the hash that the binary was built from
> > +// and will be populated by the Makefile
> > +var gitCommit = ""
> > +
> > +var versionCommand = cli.Command{
> > +	Name:        "version",
> > +	Usage:       "output the runtime version",
> > +	Description: `The version command outputs the runtime version.`,
> > +	Action:      printVersion,
> > +}
> > +
> > +func printVersion(context *cli.Context) (err error) {
> > +	if version == "" {
> 
> Like we had in the `main.go` before this change.

The code I removed from main.go was:

  - if version != "" {
  -   v = append(v, version)
  - }
  - if gitCommit != "" {
  -   v = append(v, fmt.Sprintf("commit: %s", gitCommit))
  - }
  - v = append(v, fmt.Sprintf("spec: %s", specs.Version))
  - app.Version = strings.Join(v, "\n")

That's building the output string in memory and using Join to store it
(which you need when you're using app.Version).  But with
cli.VersionPrinter, I don't think we need that single-string form
anymore, and using os.Stdout.WriteString directly avoids building the
the string in memory.  Do you want me to build the string in memory to
save some WriteString err checks?